### PR TITLE
ENH: add psdaq

### LINKED ISSUE #153: Add psdaq
https://anaconda.org/lcls-ii/psdaq

So we can do daq scans with the LCLS2 daq

Requires #87 to be resolved

### PR DESCRIPTION
**Pull request by ZLLentz**
Closes #153 

This has the code we need to do scans with the LCLS-II daq

Some caveats:
- This pulls in a lot of extra dependencies
- ~This locks us at python 3.7~
- ~This has a ton of extra cli entry points for things like starting the daq, etc., which we don't need at all~

See:
- https://github.com/slac-lcls/relmanage/blob/master/recipes/psdaq/meta.yaml
- https://github.com/slac-lcls/lcls2/blob/master/psdaq/setup.py
- https://github.com/pcdshub/pcds-recipes/tree/master/psdaq-control-minimal

### CODE REVIEW

**Comment by ZLLentz** [comment]
This seems to be bugging out on:
```
psdaq[version='>=3.3.9'] -> epics-base[version='>=7|>=7.0.4,<7.0.5.0a0']
pyca[version='>=3.1.1'] -> epics-base=3.14.12.6
```

**Comment by ZLLentz** [comment]
Offline testing with making a full environment without psdaq, then adding it in afterwards, seemed to work fine:
```
$ conda install psdaq -c lcls-ii --dry-run
Collecting package metadata (current_repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.
Solving environment: failed with repodata from current_repodata.json, will retry with next repodata source.
Collecting package metadata (repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.
Solving environment: done

## Package Plan ##

  environment location: /u1/zlentz/miniconda3/envs/test

  added / updated specs:
    - psdaq


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    amityping-1.1.5            |             py_0           6 KB  lcls-ii
    boost-1.70.0               |   py37h9de70de_1         337 KB  conda-forge
    boost-cpp-1.70.0           |       h7b93d67_3        21.1 MB  conda-forge
    cameralink-gateway-v4.9.2  |             py_0         108 KB  lcls-ii
    cython-0.29.21             |   py37hcd2ae1e_2         2.2 MB  conda-forge
    jsonpickle-1.4.1           |     pyh9f0ad1d_0          31 KB  conda-forge
    lcls2_timetool-v3.3.0      |             py_3         110 KB  lcls-ii
    libfabric-1.10.1           |       hf484d3e_1         1.1 MB  lcls-ii
    libnl-1.1.4                |       hf484d3e_0         1.5 MB  lcls-ii
    libnl3-3.2.28              |       hf484d3e_0         1.4 MB  lcls-ii
    mypy-0.790                 |             py_0         1.5 MB  conda-forge
    nose-1.3.7                 |          py_1006         118 KB  conda-forge
    p4p-3.5.2                  |   py37h10a2094_0         360 KB  conda-forge
    parse-1.18.0               |     pyh9f0ad1d_0          21 KB  conda-forge
    psalg-3.3.9                |   py37he1b5a44_0         550 KB  lcls-ii
    psdaq-3.1.8                |   py37hf484d3e_0         2.6 MB  lcls-ii
    pyserial-3.5               |     pyhd8ed1ab_0          64 KB  conda-forge
    pytest-qt-3.3.0            |             py_0          34 KB  lcls-ii
    rapidjson-1.1.0            |    he1b5a44_1002         104 KB  conda-forge
    rdma-core-13               |      hf484d3e_13         628 KB  lcls-ii
    roentdek-0.0.3             |                1          69 KB  lcls-ii
    rogue-ne-v5.2.1            |   py37hcc9cdef_0        12.5 MB  lcls-ii
    transitions-0.8.6          |     pyhd8ed1ab_0         998 KB  conda-forge
    wave8-1.0.0                |             py_0          98 KB  lcls-ii
    xtcdata-3.3.9              |       he1b5a44_0          86 KB  lcls-ii
    ------------------------------------------------------------
                                           Total:        47.6 MB

The following NEW packages will be INSTALLED:

  amityping          lcls-ii/noarch::amityping-1.1.5-py_0
  boost              conda-forge/linux-64::boost-1.70.0-py37h9de70de_1
  boost-cpp          conda-forge/linux-64::boost-cpp-1.70.0-h7b93d67_3
  cameralink-gateway lcls-ii/noarch::cameralink-gateway-v4.9.2-py_0
  cython             conda-forge/linux-64::cython-0.29.21-py37hcd2ae1e_2
  jsonpickle         conda-forge/noarch::jsonpickle-1.4.1-pyh9f0ad1d_0
  lcls2_timetool     lcls-ii/noarch::lcls2_timetool-v3.3.0-py_3
  libfabric          lcls-ii/linux-64::libfabric-1.10.1-hf484d3e_1
  libnl              lcls-ii/linux-64::libnl-1.1.4-hf484d3e_0
  libnl3             lcls-ii/linux-64::libnl3-3.2.28-hf484d3e_0
  mypy               conda-forge/noarch::mypy-0.790-py_0
  nose               conda-forge/noarch::nose-1.3.7-py_1006
  p4p                conda-forge/linux-64::p4p-3.5.2-py37h10a2094_0
  parse              conda-forge/noarch::parse-1.18.0-pyh9f0ad1d_0
  psalg              lcls-ii/linux-64::psalg-3.3.9-py37he1b5a44_0
  psdaq              lcls-ii/linux-64::psdaq-3.1.8-py37hf484d3e_0
  pyserial           conda-forge/noarch::pyserial-3.5-pyhd8ed1ab_0
  rapidjson          conda-forge/linux-64::rapidjson-1.1.0-he1b5a44_1002
  rdma-core          lcls-ii/linux-64::rdma-core-13-hf484d3e_13
  roentdek           lcls-ii/linux-64::roentdek-0.0.3-1
  rogue-ne           lcls-ii/linux-64::rogue-ne-v5.2.1-py37hcc9cdef_0
  transitions        conda-forge/noarch::transitions-0.8.6-pyhd8ed1ab_0
  wave8              lcls-ii/noarch::wave8-1.0.0-py_0
  xtcdata            lcls-ii/linux-64::xtcdata-3.3.9-he1b5a44_0

The following packages will be SUPERSEDED by a higher-priority channel:

  pytest-qt                                     conda-forge --> lcls-ii

The following packages will be DOWNGRADED:

  cairo                                1.16.0-h7979940_1007 --> 1.16.0-h9f066cc_1006
  fontconfig                           2.13.1-h736d332_1003 --> 2.13.1-h7e3eb15_1002
  harfbuzz                                 2.7.4-h5cf4720_0 --> 2.7.2-ha5b49bf_1
  icu                                       68.1-h58526e2_0 --> 67.1-he1b5a44_0
  libxml2                                 2.9.10-h72842e0_3 --> 2.9.10-h68273f3_2
  libxslt                                 1.1.33-h15afd5d_2 --> 1.1.33-hf705e74_1
  qt                                      5.12.9-h9d6b050_2 --> 5.12.9-h763d07f_1
```

**Comment by ZLLentz** [comment]
Some thoughts from Chris O'Grady that are relevant to include here:

"We should consider separating out Chris Ford’s “control” package (or a subset of it) into a lightweight conda package (ony json/zmq dependencies?) that we could both reuse as an interface, we would just agree on a version"

"As for py3.8/3.9: Valerio is working on the conda-forge “feedstock” approach which should make it easier for us to support multiple versions"

**Comment by ZLLentz** [comment]
Switching this to draft based on the above

**Comment by klauer** [comment]
Great news that they are at least considering pushing it to conda-forge! 👏

**Comment by ZLLentz** [comment]
When revisiting this, I can switch to py38 because there are now py36/37/38 uploads on the daq channel

**Comment by ZLLentz** [comment]
I'm now getting the worst merge conflict analysis I've ever seen. Thanks conda.
```
$ conda install psdaq
Collecting package metadata (current_repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.
Solving environment: failed with repodata from current_repodata.json, will retry with next repodata source.
Collecting package metadata (repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.
Solving environment: |                                                                                                                                                                         \
Found conflicts! Looking for incompatible packages.                                                                                                                                            failed

UnsatisfiableError: The following specifications were found to be incompatible with each other:

Output in format: Requested package -> Available versions

```

**Comment by ZLLentz** [comment]
It seems like there is a dependency somewhere in the chain that breaks for python=3.8, but it cannot be reported by the conda debugger for some reason

**Comment by klauer** [comment]
Oh no... Maybe that depgraph tool could provide some insight on where to look?

**Comment by ZLLentz** [comment]
I've narrowed it down to the `rogue` dependency. Chris O'Grady has contacted me offering some assistance, so I'm going to have a meeting with him and Chris Ford at some point to show me how to install a "subset" of psdaq.

**Comment by ZLLentz** [comment]
I've actually lost the ability to install latest psdaq at all on any version of python, so I'm giving up until the aforementioned meeting.

**Comment by ZLLentz** [comment]
I think something is broken unrelated to this PR, I will test this offline

**Comment by ZLLentz** [comment]
Offline, there are errors I don't understand for:
```
The following packages failed all retries:
tests/hxrsnd
tests/typhos
The following packages had race conditions, but passed at least once:
tests/hutch-python
```
But I want to merge this anyway since the environment does build and make the important psdaq.control tools available. I will debug getting the tests working in the context of the next environment tag, which I'd like to get out prior to Friday's PAMM.

File: condarc [hunk 1]
```diff
@@ -2,6 +2,7 @@
 channels:
   - conda-forge
   - pcds-tag
+  - lcls-ii
```
Comment (klauer) [inline condarc:5]: For my own understanding, we need this for `psana` and such from the new [recipe](https://github.com/pcdshub/pcds-recipes/pull/42/files)?

**Review by klauer — APPROVED** [review]

File: condarc (hunk 1 above)
Comment (ZLLentz) [reply condarc:5]: Yes, exactly. This gives us the following packages:
```
$ conda list | grep lcls-ii
amityping                 1.1.6              pyhcb9857c_4    lcls-ii
psalg                     3.3.14           py38h729150a_1    lcls-ii
psana                     3.3.14           py38h4ed09f0_1    lcls-ii
psmon                     1.0.2              pyhcb9857c_2    lcls-ii
roentdek                  0.0.3                hadeb787_2    lcls-ii
xtcdata                   3.3.14               h4ed661b_1    lcls-ii
```

**Pull request merged**